integr8ly/integreatly-operator
# MGDAPI-5732 Update prometheus alerts groups and location

# Issue link
https://issues.redhat.com/browse/MGDAPI-5732
https://issues.redhat.com/browse/MGDAPI-5731

TODO:
- tidy up imports order
- e2e once monitoring stack is in place

# What
- Moving Observability namespace alerts to redhat-rhoam-operator namespace
- Editing the alert creation logic so that alerts can be created using monitoringv1 and monitoring.rhobs
- Openshift monitoring stack remain unchanged
- Copying KCs alerts from KC namespace over to RHOAM and updating it's group to monitoring.rhobs
- Refactoring unit tests
- Updating go mod to be compatible with provided version of obo prometheus (also to be updated to most recent one in another jira as it requires rhoam deps to be updated accordingly)
- Updating probes to change the apis to monitoring.rhobs and namespace along with some unit tests refactor (the MGDAPI-5731 has been completed as part of this PR as well due to the amount of changes that would need to be replicated over both PRs)

For now, leaving along e2e tests since we don't have obo monitoring stack in place yet. 
Leaving generated-deadmansswitch alert in OO ns for now as this will need to be configured outside of RHOAM repo.

# Verification steps
- Install from this branch and confirm that alerts are created under rhoam ns - please note, OO prometheus is not a valid verification.
- Potentially wait for OBO monitoring stack before reviewing this PR
- Confirm that probes have been created under rhoam ns along with correct api group
